src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -35,12 +35,20 @@ public partial class Solution
         /// </summary>
         private AsyncLazy<Solution> CachedFrozenSolution { get; init; }
```

Comment:
now all teh cached state is *always* at the Solution level, not the SolutionCompilationState level.

Reply:
since every new SolutionCompilationState would always produce a new Solution, this was not useful to have in the inner layer (especially since everyone externally only works with the outher Solution/Document types).